client/my-sites/checkout/checkout/checkout.jsx

```diff
@@ -341,19 +343,46 @@ export class Checkout extends React.Component {
 		if ( this.props.isNewlyCreatedSite && receipt && isEmpty( receipt.failed_purchases ) ) {
 			const siteDesignType = get( selectedSite, 'options.design_type' );
 			const hasGoogleAppsInCart = cartItems.hasGoogleApps( cart );
+			const hasConciergeSessionInCart = cartItems.hasConciergeSession( cart );
 
+			// Handle the redirect path after a purchase of GSuite or Concierge Session
 			// The onboarding checklist currently supports the blog type only.
-			if ( hasGoogleAppsInCart && domainReceiptId && 'store' !== siteDesignType ) {
+			if (
+				( hasConciergeSessionInCart || ( hasGoogleAppsInCart && domainReceiptId ) ) &&
+				'store' !== siteDesignType
+			) {
 				analytics.tracks.recordEvent( 'calypso_checklist_assign', {
 					site: selectedSiteSlug,
 					plan: 'paid',
 				} );
+
+				if ( config.isEnabled( 'upsell/concierge-session' ) && hasConciergeSessionInCart ) {
+					return `/checklist/${ selectedSiteSlug }`;
+				}
 				return `/checklist/${ selectedSiteSlug }?d=gsuite`;
 			}
 
-			if ( ! hasGoogleAppsInCart && cartItems.hasDomainRegistration( cart ) ) {
+			if (
+				! hasGoogleAppsInCart &&
+				! hasConciergeSessionInCart &&
+				cartItems.hasDomainRegistration( cart )
+			) {
 				const domainsForGSuite = this.getEligibleDomainFromCart();
 
+				if ( config.isEnabled( 'upsell/concierge-session' ) ) {
```

Comment:
The issue with that approach is that we'll end up assigning a user to the test before we've decided if they're eligible for it. In other words, we might assign a user to the `show` group but then not pass the condition that checks for an eligible plan type. In that case a user would be in the `show` group but would not have seen the upsell page, and so it would pollute the test data. We need to assign an AB test group as late as possible.
I am however going to move check for `if ( domainsForGSuite.length ) {` so that we don't hit a similar case there.